databricks-sdk-java/src/main/java/com/databricks/sdk/mixin/JobsExt.java

```diff
@@ -0,0 +1,50 @@
+package com.databricks.sdk.mixin;
+
+import com.databricks.sdk.core.ApiClient;
+import com.databricks.sdk.service.jobs.GetRunRequest;
+import com.databricks.sdk.service.jobs.JobsAPI;
+import com.databricks.sdk.service.jobs.JobsService;
+import com.databricks.sdk.service.jobs.Run;
+
+public class JobsExt extends JobsAPI {
+
+  public JobsExt(ApiClient apiClient) {
+    super(apiClient);
+  }
+
+  public JobsExt(JobsService mock) {
+    super(mock);
+  }
+
+  @Override
+  public Run getRun(GetRunRequest request) {
+    Run run = super.getRun(request);
+
+    // assume that if the user sets a page token, they only want one page
+    if (request.getPageToken() != null) {
+      return run;
+    }
+
+    /*
+     * otherwise, fetch all pages and accumulate the result in a single response
+     */
+
+    boolean paginatingIterations = !run.getIterations().isEmpty();
+
+    Run currRun = run;
+    while (currRun.getNextPageToken() != null) {
+      request.setPageToken(currRun.getNextPageToken());
+      currRun = super.getRun(request);
+      if (paginatingIterations) {
+        run.getIterations().addAll(currRun.getIterations());
+      } else {
+        run.getTasks().addAll(currRun.getTasks());
+      }
```

Comment:
we need this condition because in case of For each run, the singular task keeps showing up in the tasks array?

Reply:
My understanding is that we're only ever paginating one of the two fields, and never both at the same time.
This way we avoid having duplicate entries in the array that is not being paginated.